### PR TITLE
Add an AccessNode to the ir tree as a binary parent

### PR DESCRIPTION
**Pull request by jdconrad**
The main portion of this change replaces CallNode, BraceNode, and DotNode with AccessNode as all three were just behaving as a BinaryNode parent for their respective child nodes which included a prefix node and the method call, brace access, or dot access.

This change also renames a bunch of variable names for nodes with the prefix "ir" for consistency and to differentiate them from user nodes which is helpful since both are present in many cases for the user tree to ir tree conversion.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBrace.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public AExpression getIndexNode() {
     public <Scope> void visit(UserTreeVisitor<Scope> userTreeVisitor, Scope scope) {
         userTreeVisitor.visitBrace(this, scope);
     }
-
+
```
Comment (stu-elastic) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBrace.java:54]: Remove

**Review by stu-elastic — APPROVED** [review]
:th:

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review. Will commit as soon as CI passes.

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBrace.java (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/main/java/org/elasticsearch/painless/node/EBrace.java]: Fixed.

**Pull request merged**